File: src/dataset/javascript/bob_exercism-javascript-ThomasZumsteg/openAI/ChatGPT4_bob_v2.js
class Bob {
  hey(input) {
    const trimmed = input.trim();
    if (!trimmed) return "Fine. Be that way!";
    const isQuestion = trimmed.endsWith("?");
    const isShouting = trimmed === trimmed.toUpperCase() && /[A-Z]/.test(trimmed);
    if (isShouting) return "Whoa, chill out!";
    if (isQuestion) return "Sure.";
    return "Whatever.";
  }
}

export default Bob;

File: src/dataset/cpp/bob_Exercism_cmccandless/openAI/ChatGPT4_bob_v2.cpp
#include "bob.h"
#include <cctype>

string bob::hey(string input)
{
    size_t end = input.find_last_not_of(" \t\n\r\f\v");
    if (end == string::npos) return "Fine. Be that way!";
    input = input.substr(0, end + 1);

    bool hasLetter = false;
    bool isShouting = true;

    for (char ch : input)
    {
        if (isalpha(ch))
        {
            hasLetter = true;
            if (!isupper(ch))
            {
                isShouting = false;
                break;
            }
        }
    }

    if (hasLetter && isShouting) return "Whoa, chill out!";
    if (input.back() == '?') return "Sure.";
    return "Whatever.";
}